From 045c3ebe6719f239ae9c2cde30ab42a5ee8a391d Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 10 Jul 2019 02:48:50 +0200
Subject: [PATCH 9/9] Avoid sys/sys{info,macros}.h on __APPLE__

---
 squashfs-tools/mksquashfs.c | 4 ++++
 squashfs-tools/unsquashfs.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
index e75a6c4..83ac526 100644
--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -35,7 +35,9 @@
 #include <stddef.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#if !defined(__APPLE__)
 #include <sys/sysmacros.h>
+#endif
 #include <fcntl.h>
 #include <errno.h>
 #include <dirent.h>
@@ -58,8 +60,10 @@
 #include <sys/sysctl.h>
 #else
 #include <endian.h>
+#if !defined(__APPLE__)
 #include <sys/sysinfo.h>
 #endif
+#endif
 
 #include "squashfs_fs.h"
 #include "squashfs_swap.h"
diff --git a/squashfs-tools/unsquashfs.c b/squashfs-tools/unsquashfs.c
index eb0deae..b47d887 100644
--- a/squashfs-tools/unsquashfs.c
+++ b/squashfs-tools/unsquashfs.c
@@ -32,8 +32,12 @@
 #include "stdarg.h"
 #include "fnmatch_compat.h"
 
+#ifdef __MACH__
+#include <sys/sysctl.h>
+#else
 #include <sys/sysinfo.h>
 #include <sys/sysmacros.h>
+#endif
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/resource.h>
-- 
2.20.1

